{
  "errors": [],
  "generated_at": "2024-10-27T17:38:14Z",
  "metrics": {
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\__init__.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 39,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\_catch.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 114,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\_exceptions.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 266,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\_formatting.py": {
      "CONFIDENCE.HIGH": 2,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 2,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 470,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\_suppress.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 33,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\__init__.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 0,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\apport_excepthook.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 6,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\check_types.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 20,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\conftest.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 3,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_apport_monkeypatching.py": {
      "CONFIDENCE.HIGH": 3,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 3,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 53,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py": {
      "CONFIDENCE.HIGH": 44,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 44,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 175,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py": {
      "CONFIDENCE.HIGH": 43,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 43,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 156,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_exceptions.py": {
      "CONFIDENCE.HIGH": 4,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 4,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 641,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py": {
      "CONFIDENCE.HIGH": 16,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 15,
      "SEVERITY.MEDIUM": 1,
      "SEVERITY.UNDEFINED": 0,
      "loc": 472,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_suppress.py": {
      "CONFIDENCE.HIGH": 2,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 2,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 10,
      "nosec": 0,
      "skipped_tests": 0
    },
    "_totals": {
      "CONFIDENCE.HIGH": 114,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 113,
      "SEVERITY.MEDIUM": 1,
      "SEVERITY.UNDEFINED": 0,
      "loc": 2458,
      "nosec": 0,
      "skipped_tests": 0
    }
  },
  "results": [
    {
      "code": "303                 if is_toplevel:\n304                     assert _ctx.exception_group_depth == 1\n305                     _ctx.exception_group_depth = 0\n",
      "col_offset": 20,
      "end_col_offset": 58,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 304,
      "line_range": [
        304
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "507     else:\n508         assert isinstance(exc_value, NameError)\n509         # find most recent frame\n",
      "col_offset": 8,
      "end_col_offset": 47,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\src\\exceptiongroup\\_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 508,
      "line_range": [
        508
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "3 import os\n4 import subprocess\n5 import sys\n",
      "col_offset": 0,
      "end_col_offset": 17,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_apport_monkeypatching.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Consider possible security implications associated with the subprocess module.",
      "line_number": 4,
      "line_range": [
        4
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/blacklists/blacklist_imports.html#b404-import-subprocess",
      "test_id": "B404",
      "test_name": "blacklist"
    },
    {
      "code": "30     print(\"running:\", cmd)\n31     completed = subprocess.run(\n32         cmd, env=env, stdout=subprocess.PIPE, stderr=subprocess.STDOUT\n33     )\n34     print(\"process output:\")\n",
      "col_offset": 16,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_apport_monkeypatching.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "LOW",
      "issue_text": "subprocess call - check for execution of untrusted input.",
      "line_number": 31,
      "line_range": [
        31,
        32,
        33
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b603_subprocess_without_shell_equals_true.html",
      "test_id": "B603",
      "test_name": "subprocess_without_shell_equals_true"
    },
    {
      "code": "50     file = Path(__file__).parent / \"apport_excepthook.py\"\n51     assert stdout == (\n52         f\"\"\"\\\n53   + Exception Group Traceback (most recent call last):\n54   |   File \"{file}\", line 13, in <module>\n55   |     raise ExceptionGroup(\"msg1\", [KeyError(\"msg2\"), ValueError(\"msg3\")])\n56   | exceptiongroup.ExceptionGroup: msg1 (2 sub-exceptions)\n57   +-+---------------- 1 ----------------\n58     | KeyError: 'msg2'\n59     +---------------- 2 ----------------\n60     | ValueError: msg3\n61     +------------------------------------\n62 \"\"\"\n63     )\n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_apport_monkeypatching.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 51,
      "line_range": [
        51,
        52,
        53,
        54,
        55,
        56,
        57,
        58,
        59,
        60,
        61,
        62,
        63
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "44 \n45     assert len(value_type_errors) == 2\n46 \n",
      "col_offset": 4,
      "end_col_offset": 38,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 45,
      "line_range": [
        45
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "46 \n47     assert isinstance(value_type_errors[0], ExceptionGroup)\n48     assert len(value_type_errors[0].exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 59,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 47,
      "line_range": [
        47
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "47     assert isinstance(value_type_errors[0], ExceptionGroup)\n48     assert len(value_type_errors[0].exceptions) == 1\n49     assert isinstance(value_type_errors[0].exceptions[0], ValueError)\n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 48,
      "line_range": [
        48
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "48     assert len(value_type_errors[0].exceptions) == 1\n49     assert isinstance(value_type_errors[0].exceptions[0], ValueError)\n50 \n",
      "col_offset": 4,
      "end_col_offset": 69,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 49,
      "line_range": [
        49
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "50 \n51     assert isinstance(value_type_errors[1], ExceptionGroup)\n52     assert len(value_type_errors[1].exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 59,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 51,
      "line_range": [
        51
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "51     assert isinstance(value_type_errors[1], ExceptionGroup)\n52     assert len(value_type_errors[1].exceptions) == 1\n53     assert isinstance(value_type_errors[1].exceptions[0], TypeError)\n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 52,
      "line_range": [
        52
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "52     assert len(value_type_errors[1].exceptions) == 1\n53     assert isinstance(value_type_errors[1].exceptions[0], TypeError)\n54 \n",
      "col_offset": 4,
      "end_col_offset": 68,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 53,
      "line_range": [
        53
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "54 \n55     assert len(zero_division_errors) == 1\n56     assert isinstance(zero_division_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 55,
      "line_range": [
        55
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "55     assert len(zero_division_errors) == 1\n56     assert isinstance(zero_division_errors[0], ExceptionGroup)\n57     assert isinstance(zero_division_errors[0].exceptions[0], ZeroDivisionError)\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 56,
      "line_range": [
        56
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "56     assert isinstance(zero_division_errors[0], ExceptionGroup)\n57     assert isinstance(zero_division_errors[0].exceptions[0], ZeroDivisionError)\n58     assert len(zero_division_errors[0].exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 79,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 57,
      "line_range": [
        57
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "57     assert isinstance(zero_division_errors[0].exceptions[0], ZeroDivisionError)\n58     assert len(zero_division_errors[0].exceptions) == 1\n59 \n",
      "col_offset": 4,
      "end_col_offset": 55,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 58,
      "line_range": [
        58
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "79 \n80     assert len(value_runtime_errors) == 1\n81     assert isinstance(value_runtime_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 80,
      "line_range": [
        80
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "80     assert len(value_runtime_errors) == 1\n81     assert isinstance(value_runtime_errors[0], ExceptionGroup)\n82     exceptions = value_runtime_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 81,
      "line_range": [
        81
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "82     exceptions = value_runtime_errors[0].exceptions\n83     assert isinstance(exceptions[0], ValueError)\n84     assert isinstance(exceptions[1], ValueError)\n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 83,
      "line_range": [
        83
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "83     assert isinstance(exceptions[0], ValueError)\n84     assert isinstance(exceptions[1], ValueError)\n85     assert isinstance(exceptions[2], RuntimeError)\n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 84,
      "line_range": [
        84
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "84     assert isinstance(exceptions[1], ValueError)\n85     assert isinstance(exceptions[2], RuntimeError)\n86 \n",
      "col_offset": 4,
      "end_col_offset": 50,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 85,
      "line_range": [
        85
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "86 \n87     assert len(zero_division_errors) == 1\n88     assert isinstance(zero_division_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 87,
      "line_range": [
        87
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "87     assert len(zero_division_errors) == 1\n88     assert isinstance(zero_division_errors[0], ExceptionGroup)\n89     exceptions = zero_division_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 88,
      "line_range": [
        88
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "89     exceptions = zero_division_errors[0].exceptions\n90     assert isinstance(exceptions[0], ZeroDivisionError)\n91 \n",
      "col_offset": 4,
      "end_col_offset": 55,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 90,
      "line_range": [
        90
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "106 \n107     assert len(value_runtime_errors) == 1\n108     exceptions = value_runtime_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 107,
      "line_range": [
        107
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "108     exceptions = value_runtime_errors[0].exceptions\n109     assert isinstance(exceptions[0], ValueError)\n110     assert isinstance(exceptions[1], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 109,
      "line_range": [
        109
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "109     assert isinstance(exceptions[0], ValueError)\n110     assert isinstance(exceptions[1], ExceptionGroup)\n111     assert isinstance(exceptions[1].exceptions[0], RuntimeError)\n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 110,
      "line_range": [
        110
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "110     assert isinstance(exceptions[1], ExceptionGroup)\n111     assert isinstance(exceptions[1].exceptions[0], RuntimeError)\n112 \n",
      "col_offset": 4,
      "end_col_offset": 64,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 111,
      "line_range": [
        111
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "112 \n113     assert len(zero_division_errors) == 1\n114     assert isinstance(zero_division_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 113,
      "line_range": [
        113
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "113     assert len(zero_division_errors) == 1\n114     assert isinstance(zero_division_errors[0], ExceptionGroup)\n115     assert isinstance(zero_division_errors[0].exceptions[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 114,
      "line_range": [
        114
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "114     assert isinstance(zero_division_errors[0], ExceptionGroup)\n115     assert isinstance(zero_division_errors[0].exceptions[0], ExceptionGroup)\n116     assert isinstance(\n",
      "col_offset": 4,
      "end_col_offset": 76,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 115,
      "line_range": [
        115
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "115     assert isinstance(zero_division_errors[0].exceptions[0], ExceptionGroup)\n116     assert isinstance(\n117         zero_division_errors[0].exceptions[0].exceptions[0], ZeroDivisionError\n118     )\n119 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 116,
      "line_range": [
        116,
        117,
        118
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "126     except ExceptionGroup as exc:\n127         assert exc is not group\n128     else:\n",
      "col_offset": 8,
      "end_col_offset": 31,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 127,
      "line_range": [
        127
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "156     context = exc.value.__context__\n157     assert isinstance(context, ExceptionGroup)\n158     assert str(context) == \"booboo (1 sub-exception)\"\n",
      "col_offset": 4,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 157,
      "line_range": [
        157
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "157     assert isinstance(context, ExceptionGroup)\n158     assert str(context) == \"booboo (1 sub-exception)\"\n159     assert len(context.exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 53,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 158,
      "line_range": [
        158
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "158     assert str(context) == \"booboo (1 sub-exception)\"\n159     assert len(context.exceptions) == 1\n160     assert isinstance(context.exceptions[0], ValueError)\n",
      "col_offset": 4,
      "end_col_offset": 39,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 159,
      "line_range": [
        159
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "159     assert len(context.exceptions) == 1\n160     assert isinstance(context.exceptions[0], ValueError)\n161     assert exc.value.__cause__ is None\n",
      "col_offset": 4,
      "end_col_offset": 56,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 160,
      "line_range": [
        160
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "160     assert isinstance(context.exceptions[0], ValueError)\n161     assert exc.value.__cause__ is None\n162 \n",
      "col_offset": 4,
      "end_col_offset": 38,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 161,
      "line_range": [
        161
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "180 \n181     assert len(excgrp.value.exceptions) == 2\n182     assert all(isinstance(exc, ValueError) for exc in excgrp.value.exceptions)\n",
      "col_offset": 4,
      "end_col_offset": 44,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 181,
      "line_range": [
        181
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "181     assert len(excgrp.value.exceptions) == 2\n182     assert all(isinstance(exc, ValueError) for exc in excgrp.value.exceptions)\n183     assert excgrp.value is not middle_exc\n",
      "col_offset": 4,
      "end_col_offset": 78,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 182,
      "line_range": [
        182
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "182     assert all(isinstance(exc, ValueError) for exc in excgrp.value.exceptions)\n183     assert excgrp.value is not middle_exc\n184     assert excgrp.value.__cause__ is first_exc\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 183,
      "line_range": [
        183
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "183     assert excgrp.value is not middle_exc\n184     assert excgrp.value.__cause__ is first_exc\n185     assert excgrp.value.__context__ is first_exc\n",
      "col_offset": 4,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 184,
      "line_range": [
        184
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "184     assert excgrp.value.__cause__ is first_exc\n185     assert excgrp.value.__context__ is first_exc\n186 \n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 185,
      "line_range": [
        185
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "192 \n193     assert len(lookup_errors) == 1\n194     assert isinstance(lookup_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 34,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 193,
      "line_range": [
        193
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "193     assert len(lookup_errors) == 1\n194     assert isinstance(lookup_errors[0], ExceptionGroup)\n195     exceptions = lookup_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 55,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 194,
      "line_range": [
        194
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "195     exceptions = lookup_errors[0].exceptions\n196     assert isinstance(exceptions[0], KeyError)\n197 \n",
      "col_offset": 4,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 196,
      "line_range": [
        196
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "219 \n220     assert len(excgrp.value.exceptions) == 1\n221     assert isinstance(excgrp.value.exceptions[0], KeyError)\n",
      "col_offset": 4,
      "end_col_offset": 44,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 220,
      "line_range": [
        220
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "220     assert len(excgrp.value.exceptions) == 1\n221     assert isinstance(excgrp.value.exceptions[0], KeyError)\n222     assert str(excgrp.value.exceptions[0]) == \"'foo'\"\n",
      "col_offset": 4,
      "end_col_offset": 59,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 221,
      "line_range": [
        221
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "221     assert isinstance(excgrp.value.exceptions[0], KeyError)\n222     assert str(excgrp.value.exceptions[0]) == \"'foo'\"\n",
      "col_offset": 4,
      "end_col_offset": 53,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 222,
      "line_range": [
        222
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "18 \n19     assert len(value_type_errors) == 2\n20 \n",
      "col_offset": 4,
      "end_col_offset": 38,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 19,
      "line_range": [
        19
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "20 \n21     assert isinstance(value_type_errors[0], ExceptionGroup)\n22     assert len(value_type_errors[0].exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 59,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 21,
      "line_range": [
        21
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "21     assert isinstance(value_type_errors[0], ExceptionGroup)\n22     assert len(value_type_errors[0].exceptions) == 1\n23     assert isinstance(value_type_errors[0].exceptions[0], ValueError)\n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 22,
      "line_range": [
        22
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "22     assert len(value_type_errors[0].exceptions) == 1\n23     assert isinstance(value_type_errors[0].exceptions[0], ValueError)\n24 \n",
      "col_offset": 4,
      "end_col_offset": 69,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 23,
      "line_range": [
        23
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "24 \n25     assert isinstance(value_type_errors[1], ExceptionGroup)\n26     assert len(value_type_errors[1].exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 59,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 25,
      "line_range": [
        25
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "25     assert isinstance(value_type_errors[1], ExceptionGroup)\n26     assert len(value_type_errors[1].exceptions) == 1\n27     assert isinstance(value_type_errors[1].exceptions[0], TypeError)\n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 26,
      "line_range": [
        26
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "26     assert len(value_type_errors[1].exceptions) == 1\n27     assert isinstance(value_type_errors[1].exceptions[0], TypeError)\n28 \n",
      "col_offset": 4,
      "end_col_offset": 68,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 27,
      "line_range": [
        27
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "28 \n29     assert len(zero_division_errors) == 1\n30     assert isinstance(zero_division_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 29,
      "line_range": [
        29
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "29     assert len(zero_division_errors) == 1\n30     assert isinstance(zero_division_errors[0], ExceptionGroup)\n31     assert isinstance(zero_division_errors[0].exceptions[0], ZeroDivisionError)\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 30,
      "line_range": [
        30
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "30     assert isinstance(zero_division_errors[0], ExceptionGroup)\n31     assert isinstance(zero_division_errors[0].exceptions[0], ZeroDivisionError)\n32     assert len(zero_division_errors[0].exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 79,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 31,
      "line_range": [
        31
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "31     assert isinstance(zero_division_errors[0].exceptions[0], ZeroDivisionError)\n32     assert len(zero_division_errors[0].exceptions) == 1\n33 \n",
      "col_offset": 4,
      "end_col_offset": 55,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 32,
      "line_range": [
        32
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "52 \n53     assert len(value_runtime_errors) == 1\n54     assert isinstance(value_runtime_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 53,
      "line_range": [
        53
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "53     assert len(value_runtime_errors) == 1\n54     assert isinstance(value_runtime_errors[0], ExceptionGroup)\n55     exceptions = value_runtime_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 54,
      "line_range": [
        54
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "55     exceptions = value_runtime_errors[0].exceptions\n56     assert isinstance(exceptions[0], ValueError)\n57     assert isinstance(exceptions[1], ValueError)\n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 56,
      "line_range": [
        56
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "56     assert isinstance(exceptions[0], ValueError)\n57     assert isinstance(exceptions[1], ValueError)\n58     assert isinstance(exceptions[2], RuntimeError)\n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 57,
      "line_range": [
        57
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "57     assert isinstance(exceptions[1], ValueError)\n58     assert isinstance(exceptions[2], RuntimeError)\n59 \n",
      "col_offset": 4,
      "end_col_offset": 50,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 58,
      "line_range": [
        58
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "59 \n60     assert len(zero_division_errors) == 1\n61     assert isinstance(zero_division_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 60,
      "line_range": [
        60
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "60     assert len(zero_division_errors) == 1\n61     assert isinstance(zero_division_errors[0], ExceptionGroup)\n62     exceptions = zero_division_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 61,
      "line_range": [
        61
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "62     exceptions = zero_division_errors[0].exceptions\n63     assert isinstance(exceptions[0], ZeroDivisionError)\n64 \n",
      "col_offset": 4,
      "end_col_offset": 55,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 63,
      "line_range": [
        63
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "78 \n79     assert len(value_runtime_errors) == 1\n80     exceptions = value_runtime_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 79,
      "line_range": [
        79
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "80     exceptions = value_runtime_errors[0].exceptions\n81     assert isinstance(exceptions[0], ValueError)\n82     assert isinstance(exceptions[1], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 81,
      "line_range": [
        81
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "81     assert isinstance(exceptions[0], ValueError)\n82     assert isinstance(exceptions[1], ExceptionGroup)\n83     assert isinstance(exceptions[1].exceptions[0], RuntimeError)\n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 82,
      "line_range": [
        82
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "82     assert isinstance(exceptions[1], ExceptionGroup)\n83     assert isinstance(exceptions[1].exceptions[0], RuntimeError)\n84 \n",
      "col_offset": 4,
      "end_col_offset": 64,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 83,
      "line_range": [
        83
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "84 \n85     assert len(zero_division_errors) == 1\n86     assert isinstance(zero_division_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 85,
      "line_range": [
        85
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "85     assert len(zero_division_errors) == 1\n86     assert isinstance(zero_division_errors[0], ExceptionGroup)\n87     assert isinstance(zero_division_errors[0].exceptions[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 62,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 86,
      "line_range": [
        86
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "86     assert isinstance(zero_division_errors[0], ExceptionGroup)\n87     assert isinstance(zero_division_errors[0].exceptions[0], ExceptionGroup)\n88     assert isinstance(\n",
      "col_offset": 4,
      "end_col_offset": 76,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 87,
      "line_range": [
        87
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "87     assert isinstance(zero_division_errors[0].exceptions[0], ExceptionGroup)\n88     assert isinstance(\n89         zero_division_errors[0].exceptions[0].exceptions[0], ZeroDivisionError\n90     )\n91 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 88,
      "line_range": [
        88,
        89,
        90
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "100     except ExceptionGroup as exc:\n101         assert isinstance(exc.exceptions[0], ZeroDivisionError)\n102         assert exc is not group\n",
      "col_offset": 8,
      "end_col_offset": 63,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 101,
      "line_range": [
        101
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "101         assert isinstance(exc.exceptions[0], ZeroDivisionError)\n102         assert exc is not group\n103     else:\n",
      "col_offset": 8,
      "end_col_offset": 31,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 102,
      "line_range": [
        102
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "138     context = exc.value.__context__\n139     assert isinstance(context, ExceptionGroup)\n140     assert str(context) == \"booboo (1 sub-exception)\"\n",
      "col_offset": 4,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 139,
      "line_range": [
        139
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "139     assert isinstance(context, ExceptionGroup)\n140     assert str(context) == \"booboo (1 sub-exception)\"\n141     assert len(context.exceptions) == 1\n",
      "col_offset": 4,
      "end_col_offset": 53,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 140,
      "line_range": [
        140
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "140     assert str(context) == \"booboo (1 sub-exception)\"\n141     assert len(context.exceptions) == 1\n142     assert isinstance(context.exceptions[0], ValueError)\n",
      "col_offset": 4,
      "end_col_offset": 39,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 141,
      "line_range": [
        141
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "141     assert len(context.exceptions) == 1\n142     assert isinstance(context.exceptions[0], ValueError)\n143     assert exc.value.__cause__ is None\n",
      "col_offset": 4,
      "end_col_offset": 56,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 142,
      "line_range": [
        142
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "142     assert isinstance(context.exceptions[0], ValueError)\n143     assert exc.value.__cause__ is None\n144 \n",
      "col_offset": 4,
      "end_col_offset": 38,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 143,
      "line_range": [
        143
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "152 \n153     assert len(lookup_errors) == 1\n154     assert isinstance(lookup_errors[0], ExceptionGroup)\n",
      "col_offset": 4,
      "end_col_offset": 34,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 153,
      "line_range": [
        153
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "153     assert len(lookup_errors) == 1\n154     assert isinstance(lookup_errors[0], ExceptionGroup)\n155     exceptions = lookup_errors[0].exceptions\n",
      "col_offset": 4,
      "end_col_offset": 55,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 154,
      "line_range": [
        154
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "155     exceptions = lookup_errors[0].exceptions\n156     assert isinstance(exceptions[0], KeyError)\n157 \n",
      "col_offset": 4,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 156,
      "line_range": [
        156
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "175 \n176     assert excgrp.value is not middle_exc\n177     assert excgrp.value.__cause__ is first_exc\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 176,
      "line_range": [
        176
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "176     assert excgrp.value is not middle_exc\n177     assert excgrp.value.__cause__ is first_exc\n178     assert excgrp.value.__context__ is first_exc\n",
      "col_offset": 4,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 177,
      "line_range": [
        177
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "177     assert excgrp.value.__cause__ is first_exc\n178     assert excgrp.value.__context__ is first_exc\n179 \n",
      "col_offset": 4,
      "end_col_offset": 48,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 178,
      "line_range": [
        178
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "187 \n188     assert len(excgrp.value.exceptions) == 1\n189     assert isinstance(excgrp.value.exceptions[0], KeyError)\n",
      "col_offset": 4,
      "end_col_offset": 44,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 188,
      "line_range": [
        188
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "188     assert len(excgrp.value.exceptions) == 1\n189     assert isinstance(excgrp.value.exceptions[0], KeyError)\n190     assert str(excgrp.value.exceptions[0]) == \"'foo'\"\n",
      "col_offset": 4,
      "end_col_offset": 59,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 189,
      "line_range": [
        189
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "189     assert isinstance(excgrp.value.exceptions[0], KeyError)\n190     assert str(excgrp.value.exceptions[0]) == \"'foo'\"\n",
      "col_offset": 4,
      "end_col_offset": 53,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_catch_py311.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 190,
      "line_range": [
        190
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "210         eg.add_note(\"hello\")\n211         assert eg.__notes__ == [\"hello\"]\n212         eg2 = eg.derive([ValueError()])\n",
      "col_offset": 8,
      "end_col_offset": 40,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_exceptions.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 211,
      "line_range": [
        211
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "212         eg2 = eg.derive([ValueError()])\n213         assert not hasattr(eg2, \"__notes__\")\n214 \n",
      "col_offset": 8,
      "end_col_offset": 44,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_exceptions.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 213,
      "line_range": [
        213
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "839     group = BaseExceptionGroup(\"foo\", [ValueError(1), KeyboardInterrupt()])\n840     assert repr(group) == (\n841         \"BaseExceptionGroup('foo', [ValueError(1), KeyboardInterrupt()])\"\n842     )\n843 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_exceptions.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 840,
      "line_range": [
        840,
        841,
        842
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "844     group = ExceptionGroup(\"foo\", [ValueError(1), RuntimeError(\"bar\")])\n845     assert repr(group) == \"ExceptionGroup('foo', [ValueError(1), RuntimeError('bar')])\"\n",
      "col_offset": 4,
      "end_col_offset": 87,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_exceptions.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 845,
      "line_range": [
        845
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "70     output = capsys.readouterr().err\n71     assert output == (\n72         f\"\"\"\\\n73   + Exception Group Traceback (most recent call last):\n74   |   File \"{__file__}\", line {local_lineno + 2}, in test_exceptionhook\n75   |     raise_excgroup(){underline_suffix}\n76   |   File \"{__file__}\", line {lineno + 15}, in raise_excgroup\n77   |     raise exc\n78   | {module_prefix}ExceptionGroup: test message (2 sub-exceptions)\n79   | Displays notes attached to the group too\n80   +-+---------------- 1 ----------------\n81     | Traceback (most recent call last):\n82     |   File \"{__file__}\", line {lineno + 3}, in raise_excgroup\n83     |     raise ValueError(\"foo\")\n84     | ValueError: foo\n85     +---------------- 2 ----------------\n86     | Traceback (most recent call last):\n87     |   File \"{__file__}\", line {lineno + 8}, in raise_excgroup\n88     |     raise RuntimeError(\"bar\")\n89     | RuntimeError: bar\n90     | Note from bar handler\n91     +------------------------------------\n92 \"\"\"\n93     )\n94 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 71,
      "line_range": [
        71,
        72,
        73,
        74,
        75,
        76,
        77,
        78,
        79,
        80,
        81,
        82,
        83,
        84,
        85,
        86,
        87,
        88,
        89,
        90,
        91,
        92,
        93
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "104     output = capsys.readouterr().err\n105     assert output == (\n106         f\"\"\"\\\n107   | {module_prefix}ExceptionGroup:  (1 sub-exception)\n108   +-+---------------- 1 ----------------\n109     | Exception\n110     +------------------------------------\n111 \n112 The above exception was the direct cause of the following exception:\n113 \n114 Traceback (most recent call last):\n115   File \"{__file__}\", line {lineno + 2}, in test_exceptiongroup_as_cause\n116     raise Exception() from ExceptionGroup(\"\", (Exception(),))\n117 Exception\n118 \"\"\"\n119     )\n120 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 105,
      "line_range": [
        105,
        106,
        107,
        108,
        109,
        110,
        111,
        112,
        113,
        114,
        115,
        116,
        117,
        118,
        119
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "134     output = capsys.readouterr().err\n135     assert output == (\n136         f\"\"\"\\\n137   | {module_prefix}ExceptionGroup: eg1 (1 sub-exception)\n138   +-+---------------- 1 ----------------\n139     | Exception Group Traceback (most recent call last):\n140     |   File \"{__file__}\", line {lineno}, in test_exceptiongroup_loop\n141     |     raise eg0 from eg1\n142     | {module_prefix}ExceptionGroup: eg0 (1 sub-exception)\n143     +-+---------------- 1 ----------------\n144       | Exception: e0\n145       +------------------------------------\n146 \n147 The above exception was the direct cause of the following exception:\n148 \n149   + Exception Group Traceback (most recent call last):\n150   |   File \"{__file__}\", line {lineno}, in test_exceptiongroup_loop\n151   |     raise eg0 from eg1\n152   | {module_prefix}ExceptionGroup: eg0 (1 sub-exception)\n153   +-+---------------- 1 ----------------\n154     | Exception: e0\n155     +------------------------------------\n156 \"\"\"\n157     )\n158 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 135,
      "line_range": [
        135,
        136,
        137,
        138,
        139,
        140,
        141,
        142,
        143,
        144,
        145,
        146,
        147,
        148,
        149,
        150,
        151,
        152,
        153,
        154,
        155,
        156,
        157
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "172     output = capsys.readouterr().err\n173     assert output == (\n174         f\"\"\"\\\n175   + Exception Group Traceback (most recent call last):\n176   |   File \"{__file__}\", line {local_lineno + 2}, in \\\n177 test_exceptionhook_format_exception_only\n178   |     raise_excgroup(){underline_suffix}\n179   |   File \"{__file__}\", line {lineno + 15}, in raise_excgroup\n180   |     raise exc\n181   | {module_prefix}ExceptionGroup: test message (2 sub-exceptions)\n182   | Displays notes attached to the group too\n183   +-+---------------- 1 ----------------\n184     | Traceback (most recent call last):\n185     |   File \"{__file__}\", line {lineno + 3}, in raise_excgroup\n186     |     raise ValueError(\"foo\")\n187     | ValueError: foo\n188     +---------------- 2 ----------------\n189     | Traceback (most recent call last):\n190     |   File \"{__file__}\", line {lineno + 8}, in raise_excgroup\n191     |     raise RuntimeError(\"bar\")\n192     | RuntimeError: bar\n193     | Note from bar handler\n194     +------------------------------------\n195 \"\"\"\n196     )\n197 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 173,
      "line_range": [
        173,
        174,
        175,
        176,
        177,
        178,
        179,
        180,
        181,
        182,
        183,
        184,
        185,
        186,
        187,
        188,
        189,
        190,
        191,
        192,
        193,
        194,
        195,
        196
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "200     try:\n201         exec(\"//serser\")\n202     except SyntaxError as exc:\n",
      "col_offset": 8,
      "end_col_offset": 24,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "MEDIUM",
      "issue_text": "Use of exec detected.",
      "line_number": 201,
      "line_range": [
        201
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b102_exec_used.html",
      "test_id": "B102",
      "test_name": "exec_used"
    },
    {
      "code": "214     output = capsys.readouterr().err\n215     assert output == (\n216         f\"\"\"\\\n217 Traceback (most recent call last):\n218   File \"{__file__}\", line {lineno + 2}, \\\n219 in test_formatting_syntax_error\n220     exec(\"//serser\"){underline_suffix}\n221   File \"<string>\", line 1\n222     //serser{underline}\n223 SyntaxError: invalid syntax\n224 \"\"\"\n225     )\n226 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 215,
      "line_range": [
        215,
        216,
        217,
        218,
        219,
        220,
        221,
        222,
        223,
        224,
        225
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "261         lineno = raise_excgroup.__code__.co_firstlineno\n262         assert isinstance(lines, list)\n263         module_prefix = \"\" if sys.version_info >= (3, 11) else \"exceptiongroup.\"\n",
      "col_offset": 8,
      "end_col_offset": 38,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 262,
      "line_range": [
        262
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "266         )\n267         assert \"\".join(lines) == (\n268             f\"\"\"\\\n269   + Exception Group Traceback (most recent call last):\n270   |   File \"{__file__}\", line {local_lineno + 25}, in test_format_exception\n271   |     raise_excgroup(){underline_suffix}\n272   |   File \"{__file__}\", line {lineno + 15}, in raise_excgroup\n273   |     raise exc\n274   | {module_prefix}ExceptionGroup: test message (2 sub-exceptions)\n275   | Displays notes attached to the group too\n276   +-+---------------- 1 ----------------\n277     | Traceback (most recent call last):\n278     |   File \"{__file__}\", line {lineno + 3}, in raise_excgroup\n279     |     raise ValueError(\"foo\")\n280     | ValueError: foo\n281     +---------------- 2 ----------------\n282     | Traceback (most recent call last):\n283     |   File \"{__file__}\", line {lineno + 8}, in raise_excgroup\n284     |     raise RuntimeError(\"bar\")\n285     | RuntimeError: bar\n286     | Note from bar handler\n287     +------------------------------------\n288 \"\"\"\n289         )\n290 \n",
      "col_offset": 8,
      "end_col_offset": 9,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 267,
      "line_range": [
        267,
        268,
        269,
        270,
        271,
        272,
        273,
        274,
        275,
        276,
        277,
        278,
        279,
        280,
        281,
        282,
        283,
        284,
        285,
        286,
        287,
        288,
        289
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "323     raise_exc_lineno3 = raise_exc.__code__.co_firstlineno + 7\n324     assert isinstance(lines, list)\n325     assert \"\".join(lines) == (\n",
      "col_offset": 4,
      "end_col_offset": 34,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 324,
      "line_range": [
        324
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "324     assert isinstance(lines, list)\n325     assert \"\".join(lines) == (\n326         f\"\"\"\\\n327 Traceback (most recent call last):\n328   File \"{__file__}\", line {raise_exc_lineno2}, in raise_exc\n329     raise_exc(max_level, level + 1){underline_suffix1}\n330   File \"{__file__}\", line {raise_exc_lineno1}, in raise_exc\n331     raise Exception(f\"LEVEL_{{level}}\")\n332 Exception: LEVEL_3\n333 \n334 During handling of the above exception, another exception occurred:\n335 \n336 Traceback (most recent call last):\n337   File \"{__file__}\", line {raise_exc_lineno2}, in raise_exc\n338     raise_exc(max_level, level + 1){underline_suffix1}\n339   File \"{__file__}\", line {raise_exc_lineno3}, in raise_exc\n340     raise Exception(f\"LEVEL_{{level}}\")\n341 Exception: LEVEL_2\n342 \n343 During handling of the above exception, another exception occurred:\n344 \n345 Traceback (most recent call last):\n346   File \"{__file__}\", line {local_lineno}, in test_format_nested\n347     raise_exc(3){underline_suffix2}\n348   File \"{__file__}\", line {raise_exc_lineno3}, in raise_exc\n349     raise Exception(f\"LEVEL_{{level}}\")\n350 Exception: LEVEL_1\n351 \"\"\"\n352     )\n353 \n",
      "col_offset": 4,
      "end_col_offset": 5,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 325,
      "line_range": [
        325,
        326,
        327,
        328,
        329,
        330,
        331,
        332,
        333,
        334,
        335,
        336,
        337,
        338,
        339,
        340,
        341,
        342,
        343,
        344,
        345,
        346,
        347,
        348,
        349,
        350,
        351,
        352
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "375         module_prefix = \"\" if sys.version_info >= (3, 11) else \"exceptiongroup.\"\n376         assert output == [\n377             f\"{module_prefix}ExceptionGroup: test message (2 sub-exceptions)\\n\",\n378             \"Displays notes attached to the group too\\n\",\n379         ]\n380 \n",
      "col_offset": 8,
      "end_col_offset": 9,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 376,
      "line_range": [
        376,
        377,
        378,
        379
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "408         output = capsys.readouterr().err\n409         assert output == (\n410             f\"\"\"\\\n411   + Exception Group Traceback (most recent call last):\n412   |   File \"{__file__}\", line {local_lineno + 13}, in test_print_exception\n413   |     raise_excgroup(){underline_suffix}\n414   |   File \"{__file__}\", line {lineno + 15}, in raise_excgroup\n415   |     raise exc\n416   | {module_prefix}ExceptionGroup: test message (2 sub-exceptions)\n417   | Displays notes attached to the group too\n418   +-+---------------- 1 ----------------\n419     | Traceback (most recent call last):\n420     |   File \"{__file__}\", line {lineno + 3}, in raise_excgroup\n421     |     raise ValueError(\"foo\")\n422     | ValueError: foo\n423     +---------------- 2 ----------------\n424     | Traceback (most recent call last):\n425     |   File \"{__file__}\", line {lineno + 8}, in raise_excgroup\n426     |     raise RuntimeError(\"bar\")\n427     | RuntimeError: bar\n428     | Note from bar handler\n429     +------------------------------------\n430 \"\"\"\n431         )\n432 \n",
      "col_offset": 8,
      "end_col_offset": 9,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 409,
      "line_range": [
        409,
        410,
        411,
        412,
        413,
        414,
        415,
        416,
        417,
        418,
        419,
        420,
        421,
        422,
        423,
        424,
        425,
        426,
        427,
        428,
        429,
        430,
        431
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "456         output = capsys.readouterr().err\n457         assert output == (\n458             f\"\"\"\\\n459   + Exception Group Traceback (most recent call last):\n460   |   File \"{__file__}\", line {local_lineno + 13}, in test_print_exc\n461   |     raise_excgroup(){underline_suffix}\n462   |   File \"{__file__}\", line {lineno + 15}, in raise_excgroup\n463   |     raise exc\n464   | {module_prefix}ExceptionGroup: test message (2 sub-exceptions)\n465   | Displays notes attached to the group too\n466   +-+---------------- 1 ----------------\n467     | Traceback (most recent call last):\n468     |   File \"{__file__}\", line {lineno + 3}, in raise_excgroup\n469     |     raise ValueError(\"foo\")\n470     | ValueError: foo\n471     +---------------- 2 ----------------\n472     | Traceback (most recent call last):\n473     |   File \"{__file__}\", line {lineno + 8}, in raise_excgroup\n474     |     raise RuntimeError(\"bar\")\n475     | RuntimeError: bar\n476     | Note from bar handler\n477     +------------------------------------\n478 \"\"\"\n479         )\n480 \n",
      "col_offset": 8,
      "end_col_offset": 9,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 457,
      "line_range": [
        457,
        458,
        459,
        460,
        461,
        462,
        463,
        464,
        465,
        466,
        467,
        468,
        469,
        470,
        471,
        472,
        473,
        474,
        475,
        476,
        477,
        478,
        479
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "502         output = capsys.readouterr().err\n503         assert \"Did you mean\" in output and \"'filter'?\" in output\n504 \n",
      "col_offset": 8,
      "end_col_offset": 65,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 503,
      "line_range": [
        503
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "527         output = capsys.readouterr().err\n528         assert \"Did you mean\" in output and \"'append'?\" in output\n529 \n",
      "col_offset": 8,
      "end_col_offset": 65,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 528,
      "line_range": [
        528
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "551         output = capsys.readouterr().err\n552         assert \"NamedAttributeError\" in output\n553 \n",
      "col_offset": 8,
      "end_col_offset": 46,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_formatting.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 552,
      "line_range": [
        552
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "14 \n15     assert len(exc.value.exceptions) == 1\n16     assert isinstance(exc.value.exceptions[0], RuntimeError)\n",
      "col_offset": 4,
      "end_col_offset": 41,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_suppress.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 15,
      "line_range": [
        15
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "15     assert len(exc.value.exceptions) == 1\n16     assert isinstance(exc.value.exceptions[0], RuntimeError)\n",
      "col_offset": 4,
      "end_col_offset": 60,
      "filename": "D:\\bandit\\upstream-repos\\py3-exceptiongroup\\tests\\test_suppress.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 16,
      "line_range": [
        16
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    }
  ]
}